spacy/training/loop.py

```diff
@@ -198,6 +201,9 @@ def train_while_improving(
     words_seen = 0
     start_time = timer()
     for step, (epoch, batch) in enumerate(train_data):
+        if before_update:
+            before_update_args = {"current_step": step}
```

Comment:
Maybe pass through the epoch as well?

Reply:
Also, why `current_stop` over simply `step`?

Reply:
Re. `current_step`: No particular reason; I think I copy-pasted that from elsewhere. Will keep it as `step` for consistency.